core/src/avm2/globals/flash/display/NativeWindow.as

```diff
@@ -0,0 +1,314 @@
+  {
+    public const systemMaxSize:Point = new Point(2880, 2880);
+    public const systemMinSize:Point = new Point(1, 1);
+    public var minSize:Point = systemMinSize;
+    public var maxSize:Point = systemMaxSize;
+    public var title:String;
+    public var alwaysInFront:Boolean = true;
+    public var visible:Boolean = true;
+
+    private var _bounds:Rectangle;
+    private var _maximizable:Boolean;
+    private var _minimizable:Boolean;
+    private var _resizable:Boolean;
+    private var _systemChrome:String;
+    private var _transparent:Boolean;
+    private var _type:String;
+    private var _closed:Boolean = false;
+    private var _stage:Stage;
+
+    public function NativeWindow(initOptions:NativeWindowInitOptions, _stage:Stage = null)
+    {
+      stub_constructor("flash.display.NativeWindow");
+      NativeApplication.nativeApplication._openedWindows.push(this);
+      if (_stage)
+      {
+        this._stage = _stage;
+        _stage.addEventListener(Event.RESIZE, function(e:Event):void
+        {
+          dispatchEvent(new NativeWindowBoundsEvent(NativeWindowBoundsEvent.RESIZE, false, false, _bounds, _bounds = new Rectangle(x, y, width, height)));
+        });
+      }
+
+      _maximizable = initOptions.maximizable;
+      _minimizable = initOptions.minimizable;
+      _resizable = initOptions.resizable;
+      _systemChrome = initOptions.systemChrome;
+      _transparent = initOptions.transparent;
+      _type = initOptions.type;
+    }
+
+    public function get width():Number
+    {
+      stub_getter("flash.display.NativeWindow", "width");
+      return _stage.stageWidth;
+    }
+
+    public function set width(value:Number):void
+    {
+      stub_setter("flash.display.NativeWindow", "width");
+      _stage.stageWidth = value;
+    }
+
+    public function get height():Number
+    {
+      stub_getter("flash.display.NativeWindow", "height");
+      return _stage.stageHeight;
+    }
+
+    public function set height(value:Number):void
+    {
+      stub_setter("flash.display.NativeWindow", "height");
+      _stage.stageHeight = value;
+    }
+
+    public function get x():Number
+    {
+      stub_getter("flash.display.NativeWindow", "x");
+      return _stage.x;
+    }
+
+    public function set x(value:Number):void
+    {
+      stub_setter("flash.display.NativeWindow", "x");
+    }
+
+    public function get y():Number
+    {
+      stub_getter("flash.display.NativeWindow", "y");
+      return _stage.y;
+    }
+
+    public function set y(value:Number):void
+    {
+      stub_setter("flash.display.NativeWindow", "y");
+    }
+
+    public function get bounds():Rectangle
+    {
+      stub_getter("flash.display.NativeWindow", "bounds");
+      return _bounds;
+    }
+
+    public function set bounds(value:Rectangle):void
+    {
+      stub_setter("flash.display.NativeWindow", "bounds");
+      _bounds.height = height = value.height;
+      _bounds.width = width = value.width;
```

Comment:
Why set `height` and `width`, but not `x` and `y`?

Also, I feel like invoking other setters (and getters too) of other properties with which there is this tight coupling, might not be the best idea, as it can easily lead to infinite (indirect) recursion.

And, maybe `set x`, `set y`, `set width`, and `set height` should also affect the bounds, to keep things consistent - but again, perhaps directly via `_bounds` to avoid infinite property accessor recursion.

Reply:
What do you recommend to do instead?

Reply:
> Setting the bounds property of a window is equivalent to setting its x, y, width, and height properties. Likewise, setting any of the individual dimensions is equivalent to setting the bounds property. When you set all the dimensions at the same time by using the bounds property, **fewer events are dispatched.**

Should we dispatch these events too, and to which events the docs refer?

Reply:
I'd say `set bounds` should simply set `_bounds` - but probably still copying all four values within individually, to not "steal" the `Rectangle` object reference; and not touch any other setters of `this`.
And the 4 individual setters should also go directly into the appropriate member of `_bounds`.
Finally, the setters of `width`, `height`, and `bounds` could also set the size of the stage.
Perhaps this way, everything will be consistent, and there's no risk of infinite recursion.

Reply:
> Should we dispatch these events too, and to which events the docs refer?

I don't have any strong opinions: Since these are just stubs, I don't think event dispatching in general is that important yet, as nothing will work anyway.